src/components/events/EventCard.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/Card';
+
+import { Avatar, AvatarImage } from '@/components/ui/Avatar';
+import { Badge } from '@/components/ui/Badge';
+import { Link } from '@/lib/locale/navigation';
+import { cx } from '@/lib/utils';
+import { getTranslations } from 'next-intl/server';
+
+type EventCardProps = {
+  id: number;
+  title: string;
+  subheader: string;
+  description: string;
+  imagePath?: string;
+  startTime: Date;
+  endTime: Date;
+  internal: boolean;
+  locale: string;
+  wrapperClassName?: string;
+  cardClassName?: string;
+  _active?: boolean;
+};
+
+/**
+ * A card for an event.
+ * Only set the _active prop to true if you're testing active events.
+ */
+async function EventCard(props: EventCardProps) {
+  const t = await getTranslations('events');
+  const tUi = await getTranslations('ui');
+
+  const dateOptions = {
+    hour: '2-digit',
+    minute: '2-digit',
+    hour12: false,
+  } as const;
+
+  // For example "18:00", same for end time
+  const formattedStartTime = props.startTime.toLocaleTimeString(
+    props.locale,
+    dateOptions,
+  );
+  // For example "22/8/2024" or "22.8.2024", same for end date
+  const formattedStartDate = props.startTime.toLocaleDateString(props.locale);
+  const formattedEndTime = props.endTime.toLocaleTimeString(
+    props.locale,
+    dateOptions,
+  );
+  const formattedEndDate = props.endTime.toLocaleDateString(props.locale);
+
+  const started = props.startTime < new Date() || props._active;
+  const ended = props.endTime < new Date();
+
+  return (
+    <Link
+      href={`/events/${props.id}`}
+      aria-label={t('detailsAboutEvent', { eventName: props.title })}
+      className={props.wrapperClassName}
```

Comment:
do we need a different className prop for the link and the card? The Link will mostly be static so i think only having it on the card is fine. Also in general it is not necessary if we only need to define it once. Then we can just set the classes within the component

Reply:
Not all event cards need the same css classes, so having optional classnames as the props would be a good solution